pandas/core/reshape/pivot.py

```diff
@@ -683,3 +699,57 @@ def _get_names(arrs, names, prefix="row"):
             names = list(names)
 
     return names
+
+
+def _build_names_mapper(
+    names: List[str], shared_col_row_names: Set[str], suffix: str
+) -> Tuple[Dict[str, str], List[str]]:
+    """
+    Given a list of row or column names, creates a mapper of unique names to
+    column/row names.
+
+    Parameters
+    ----------
+    names : list
+        Names to be deduplicated.
+    shared_col_row_names : set or list
+        Values used both in rows and columns, so need additional deduplication.
+    suffix : str
+        Suffix to deduplicate values in shared_col_row_names
+
+    Returns
+    -------
+    names_mapper: dict
+        The keys are the unique names and the values are the original names.
+    unique_names: list
+        Unique names in the same order that names came in
+    """
```

Comment:
seems like you can use Counter here no?

Reply:
Yes, is that preferred over _value_counts_arraylike?